### PR TITLE
Fugue for PyCaret3

### LINKED ISSUE #2452: [ENH]: Add fugue support for 3.0.0rc
### Describe the feature you want to add to this project

Incorporate fugue/spark parallelization for `compare_models` in 3.0.0rc release

### Describe your proposed solution

- [ ] Fugue support for 2.3.10 will have to be reworked for 3.0.0 because of new OOP structure.
- [ ] Also add it to the `time_series` module which did not have this capability earlier.

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by goodwanghan**
## Related Issue or bug

https://github.com/pycaret/pycaret/issues/2452

Closes #2452 

#### Describe the changes you've made

Add Fugue parallel backend to PyCaret 3

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Not tested yet

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)

NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by ngupta23 — COMMENTED** [review]
Could you delete AUC.png? It is probably created when you run the tests.

**Comment by goodwanghan** [comment]
> Could you delete AUC.png? It is probably created when you run the tests.

Done

**Review by Yard1 — CHANGES_REQUESTED** [review]
Thanks, this is great! I have a few comments.

File: pycaret/internal/parallel/parallel_backend.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from typing import Any, Dict, List, Optional, Type, Union
+
+from pycaret.internal.display import CommonDisplay
+
+
+class NoDisplay(CommonDisplay):
+    """The Display class to completely turn off all displays"""
+
+    def can_display(self, override):
+        return False
```
[the remark this answers is not in the document]
Comment (goodwanghan) [reply pycaret/internal/parallel/parallel_backend.py]: DOne

File: pycaret/classification/oop.py [hunk 2]
```diff
@@ -926,6 +941,8 @@ def compare_models(
         experiment_custom_tags: Optional[Dict[str, Any]] = None,
         probability_threshold: Optional[float] = None,
         verbose: bool = True,
+        display: Optional[CommonDisplay] = None,
```
[the remark this answers is not in the document]
Comment (goodwanghan) [reply pycaret/classification/oop.py]: Removed

File: pycaret/internal/pycaret_experiment/pycaret_experiment.py [hunk 3]
```diff
@@ -38,6 +38,19 @@ def __init__(self) -> None:
...
+        return {"_setup_params": self._setup_params}
+
+    def _unpack_at_remote(self, data: dict) -> None:
+        for k, v in data.items():
+            setattr(self, k, v)
+
+    def _register_setup_params(self, params: dict) -> None:
+        self._setup_params = {
+            k: v for k, v in params.items() if k != "self" and v is not None
+        }
```
[the remark this answers is not in the document]
Comment (goodwanghan) [reply pycaret/internal/pycaret_experiment/pycaret_experiment.py]: Done

File: pycaret/classification/oop.py [hunk 4]
```diff
@@ -613,10 +618,20 @@ def setup(
 
         """
 
+        self._register_setup_params(dict(locals()))
+
+        assert (data is None and data_func is not None) or (
```
[the remark this answers is not in the document]
Comment (goodwanghan) [reply pycaret/classification/oop.py]: Done

**Pull request merged**